Remotion/Web/Development.WebTesting/HostingStrategies/DockerHosting/DockerHelper.cs

```diff
@@ -0,0 +1,199 @@
+    }
+
+    public void Pull (string dockerImageName)
+    {
+      ArgumentUtility.CheckNotNullOrEmpty ("dockerImageName", dockerImageName);
+
+      var dockerPullCommand = $"pull {dockerImageName}";
+
+      RunDockerCommand (dockerPullCommand);
+    }
+
+    public void Build (string tag, IReadOnlyDictionary<string, string> buildArgs, IDockerFile dockerFile)
+    {
+      ArgumentUtility.CheckNotNullOrEmpty ("tag", tag);
+      ArgumentUtility.CheckNotNull ("buildArgs", buildArgs);
+      ArgumentUtility.CheckNotNull ("dockerFile", dockerFile);
+
+      var buildContextPath = Path.GetDirectoryName (dockerFile.GetDockerFileFullPath());
+
+      var buildDockerImageCommand =
+          "build"
+          + $" -t {tag}"
+          + string.Join ("", buildArgs.Select (x => $" --build-arg {x.Key}={x.Value}"))
+          + $" \"{buildContextPath}\"";
+
+      RunDockerCommand (buildDockerImageCommand, buildContextPath);
+    }
+
+    public void Run (
+        bool detached,
+        bool removeContainer,
+        IReadOnlyDictionary<int, int> publishedPorts,
+        string containerName,
+        string hostName,
+        string imageName)
+    {
+      ArgumentUtility.CheckNotNull ("publishedPorts", publishedPorts);
+      ArgumentUtility.CheckNotNullOrEmpty ("containerName", containerName);
+      ArgumentUtility.CheckNotEmpty ("hostName", hostName);
+      ArgumentUtility.CheckNotNullOrEmpty ("imageName", imageName);
+
+      var iisHostWebSiteInDockerCommand =
+          "run"
+          + (detached ? " -d" : "")
+          + (removeContainer ? " --rm" : "")
+          + string.Join ("", publishedPorts.Select (x => $" -p {x.Key}:{x.Value}"))
+          + $" --name {containerName}"
+          + (hostName != null ? $" --hostname {hostName}" : "")
+          + $" {imageName}";
+
+      RunDockerCommand (iisHostWebSiteInDockerCommand);
+
+      var verifyContainerIsStartedCommand =
+          "inspect"
+          + " -f '{{.State.Running}}'"
+          + $" {containerName}";
+
+      while (_lastCommandLineOutputLine != "'true'")
+        RunDockerCommand (verifyContainerIsStartedCommand);
+    }
+
+    public void Stop (string containerName)
+    {
+      ArgumentUtility.CheckNotNullOrEmpty ("containerName", containerName);
+
+      var stopDockerContainerCommand = $"stop {containerName}";
+      RunDockerCommand (stopDockerContainerCommand);
+    }
+
+    public void RemoveImage (string imageName)
+    {
+      ArgumentUtility.CheckNotNullOrEmpty ("imageName", imageName);
+
+      var removeDockerImageCommand = $"image rm {imageName}";
+      RunDockerCommand (removeDockerImageCommand);
+    }
+
+    private string GetDockerExeFullPath ()
+    {
+      //Note: We explicitly do not use Environment.ProgramFiles, as it chooses ProgramFiles (x86) when we run our Unit Tests in a 32 Bit Context
+      // As the Build Server is 64 bit, Docker cannot be found there under these circumstances.
+      const string programFiles = "C:\\Program Files";

```

Comment:
Still don't believe there's a reason not to use SpecialPaths instead of hardcoded strings. See browser infrastructure.